File: frontend/src/api/quotation.ts
import request from './request'

export interface QuotationItem {
  country: string | null
  name: string
  standard: string | null
  lr_or_not: string | null
  weeks: number | null
  local_testing: string | null
  models: string | null
  unit_price: number
  discount: number
  amount: number
  item_remark: string | null
}

export interface QuotationListItem {
  id: string
  quote_no: string
  opp_id: string | null
  customer_id: string | null
  version: number
  total_amount: number
  discount_amount: number | null
  discount_rate: number | null
  valid_until: string
  status: string
  created_by: string | null
  created_at: string
}

export interface QuotationDetail extends QuotationListItem {
  items: QuotationItem[]
  contact_name: string | null
  contact_phone: string | null
  deliver_to_address: string | null
  payment_terms: string | null
  remark: string | null
  approved_by: string | null
  updated_at: string
}

export const quotationApi = {
  list(params: { page?: number; page_size?: number; status?: string; opp_id?: string }) {
    return request.get<any, { code: number; data: QuotationListItem[]; total: number }>('/quotations', { params })
  },
  get(id: string) {
    return request.get<any, { code: number; data: QuotationDetail }>(`/quotations/${id}`)
  },
  create(data: Record<string, unknown>) {
    return request.post<any, { code: number; data: QuotationDetail }>('/quotations', data)
  },
  update(id: string, data: Record<string, unknown>) {
    return request.put<any, { code: number; data: QuotationDetail }>(`/quotations/${id}`, data)
  },
  delete(id: string) {
    return request.delete(`/quotations/${id}`)
  },
}

File: backend/app/schemas/quotation.py
import uuid
from datetime import datetime, date
from decimal import Decimal
from pydantic import BaseModel, ConfigDict


class QuotationItem(BaseModel):
    country: str | None = None          # 国家
    name: str                           # 认证名称/项目
    standard: str | None = None         # 认证标准
    lr_or_not: str | None = None        # 是否需要当地代表LR: Y/N
    local_testing: str | None = None    # 是否本地测试: Y/N
    models: str | None = None           # 型号/证书数量描述
    weeks: int | None = None            # 认证周期（周）
    unit_price: Decimal
    discount: Decimal = Decimal("1.0")
    amount: Decimal
    item_remark: str | None = None      # 明细备注


class QuotationBase(BaseModel):
    opp_id: uuid.UUID | None = None
    customer_id: uuid.UUID | None = None
    items: list[QuotationItem] = []
    total_amount: Decimal
    discount_amount: Decimal | None = None
    discount_rate: Decimal | None = None
    valid_until: date
    status: str = "草稿"
    contact_name: str | None = None
    contact_phone: str | None = None
    deliver_to_address: str | None = None
    product_name: str | None = None
    product_model: str | None = None
    payment_terms: str | None = None
    remark: str | None = None


class QuotationCreate(QuotationBase):
    pass


class QuotationUpdate(BaseModel):
    opp_id: uuid.UUID | None = None
    customer_id: uuid.UUID | None = None
    items: list[QuotationItem] | None = None
    total_amount: Decimal | None = None
    discount_amount: Decimal | None = None
    discount_rate: Decimal | None = None
    valid_until: date | None = None
    status: str | None = None
    contact_name: str | None = None
    contact_phone: str | None = None
    deliver_to_address: str | None = None
    product_name: str | None = None
    product_model: str | None = None
    payment_terms: str | None = None
    remark: str | None = None


class QuotationOut(QuotationBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    quote_no: str
    version: int
    created_by: uuid.UUID | None
    approved_by: uuid.UUID | None
    created_at: datetime
    updated_at: datetime


class QuotationListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    quote_no: str
    opp_id: uuid.UUID | None
    customer_id: uuid.UUID | None
    version: int
    total_amount: Decimal
    discount_amount: Decimal | None
    discount_rate: Decimal | None
    valid_until: date
    status: str
    created_by: uuid.UUID | None
    created_at: datetime
